calico_containers/pycalico/netns.py

```diff
@@ -182,23 +182,29 @@ def set_veth_mac(veth_name_host, mac):
                timeout=IP_CMD_TIMEOUT)
 
 
-def add_ns_default_route(namespace, next_hop, veth_name_ns):
+def add_ns_default_route(namespace, ip_version, veth_name_ns):
     """
     Add a default route to the namespace.
 
     :param namespace: The namespace to operate in.
     :type namespace Namespace
-    :param next_hop: The next hop IP used as the default route in the namespace.
+    :param ip_version: The IP version - '4' or '6'
     :param veth_name_ns: The name of the interface in the namespace.
     :return: None. Raises CalledProcessError on error.
     """
-    assert isinstance(next_hop, IPAddress)
     with NamedNamespace(namespace) as ns:
+        # Use the first link local address as the dummy address.
+        # This only ever appears in an ARP packet.
+        # The host has proxy arp configured on the other end of the veth so
+        # will respond with the MAC of the veth
+        dummy_next_hop_4 = "169.254.0.0"
```

Comment:
This one isn't really a link local address: while 169.254/16 is generally used for link local addresses, the first and last 256 addresses are somehow special.  See https://tools.ietf.org/html/rfc3927 section 8.

.1.0 seems safer.